packages/typespec-test/test/healthInsights_radiologyinsights/generated/typespec-ts/src/models/fhir/r4/models.ts

```diff
@@ -1077,7 +1125,17 @@ export interface DomainResource extends Resource {
 
 export function domainResourceDeserializer(item: any): DomainResource {
   return {
-    ...item,
+    additionalProperties: serializeRecord(item, [
```

Comment:
For output, can we build some mapper between the knownProperties name and value ser/deserializer and make this part more generic?

Reply:
For input, we do that what is in the additionalProperties bag are user intend additional properties, knownProperties is unnecessary, if they have the same property name but with different value type conflicts, we should report warnings at runtime.

Reply:
I may not quite understand the input part but for output if we have that case the knownable property would override the additional property ones. This is simple and straightforward.